pkg/config/validation/validation.go

```diff
@@ -1025,6 +1025,18 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 					}
 				}
 			}
+
+			if i.Tls != nil {
+				if i.Tls.Mode == networking.ServerTLSSettings_ISTIO_MUTUAL || i.Tls.Mode == networking.ServerTLSSettings_AUTO_PASSTHROUGH {
+					errs = appendValidation(errs, fmt.Errorf("configuration is invalid: cannot set mode to %s in sidecar ingress tls", i.Tls.Mode.String()))
+				}
+				protocol := protocol.Parse(i.Port.Protocol)
+				if !protocol.IsTLS() {
+					errs = appendValidation(errs, fmt.Errorf("server cannot have TLS settings for non HTTPS/TLS ports"))
+				}
+				errs = appendValidation(errs, validateTLSOptions(i.Tls))
```

Comment:
Should also reject `httpsRedirect`? or support it, but would prefer to reject.

`subjectAltNames` - should probably also be rejected. I don't think it makes much sense to match on SAN when we only allow a single TLS config to apply (and thus there is no SNI routing, unlike at gateway).

Reply:
@howardjohn ... added the following validations